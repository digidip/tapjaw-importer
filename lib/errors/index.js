"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.isError = exports.TapjawConfigError = exports.ConnectorError = exports.AdapterError = void 0;
var adapter_error_1 = require("./adapter-error");
Object.defineProperty(exports, "AdapterError", { enumerable: true, get: function () { return __importDefault(adapter_error_1).default; } });
var connector_error_1 = require("./connector-error");
Object.defineProperty(exports, "ConnectorError", { enumerable: true, get: function () { return __importDefault(connector_error_1).default; } });
var tapjaw_config_error_1 = require("./tapjaw-config-error");
Object.defineProperty(exports, "TapjawConfigError", { enumerable: true, get: function () { return __importDefault(tapjaw_config_error_1).default; } });
var is_error_1 = require("./typeguards/is-error");
Object.defineProperty(exports, "isError", { enumerable: true, get: function () { return __importDefault(is_error_1).default; } });
