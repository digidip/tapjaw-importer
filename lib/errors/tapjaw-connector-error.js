"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const is_error_1 = tslib_1.__importDefault(require("../typeguards/is-error"));
class TapjawConnectorError extends Error {
    constructor(message, connector) {
        if (typeof message === 'string') {
            super(`${connector.constructor.name}: ${message}`);
            this.name = `TapjawConnectorError:${connector.constructor.name}`;
        }
        else if ((0, is_error_1.default)(message)) {
            super(`${connector.constructor.name}: ${String(message)}`);
            this.name = `TapjawConnectorError:${connector.constructor.name}`;
            this.parentStack = message.stack;
        }
        else {
            throw new Error('Unsupported message data type passed to TapjawConnectorError');
        }
    }
}
exports.default = TapjawConnectorError;
