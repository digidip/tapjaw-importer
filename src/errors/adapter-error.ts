import TapjawAdapter from '../contracts/tapjaw-adapter';
import TapjawMessage from '../contracts/tapjaw-message';
import isError from './typeguards/is-error';

export default class AdapterError extends Error {
    public parentStack?: string;

    constructor(message: unknown, adapter: TapjawAdapter<unknown, TapjawMessage>) {
        if (typeof message === 'string') {
            super(message);
            this.name = `AdapterError:${adapter.constructor.name}`;
        } else if (isError(message)) {
            super(String(message));
            this.name = `AdapterError:${adapter.constructor.name}`;
            this.parentStack = message.stack;
        } else {
            throw new Error('Unsupported message data type passed to AdapterError');
        }
    }
}
