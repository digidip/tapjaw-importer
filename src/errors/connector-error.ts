import TapjawConnector from '../contracts/tapjaw-connector';
import isError from './typeguards/is-error';

export default class ConnectorError extends Error {
    public parentStack?: string;

    constructor(message: unknown, adapter: TapjawConnector) {
        if (typeof message === 'string') {
            super(message);
            this.name = `ConnectorError:${adapter.constructor.name}`;
        } else if (isError(message)) {
            super(String(message));

            this.name = `ConnectorError:${adapter.constructor.name}`;
            this.parentStack = message.stack;
        } else {
            throw new Error('Unsupported message data type passed to ConnectorError');
        }
    }
}
