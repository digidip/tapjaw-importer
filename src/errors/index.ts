export { default as TapjawAdapterError } from './tapjaw-adapter-error';
export { default as TapjawConnectorError } from './tapjaw-connector-error';
export { default as TapjawConfigError } from './tapjaw-config-error';
export { default as isError } from './typeguards/is-error';
