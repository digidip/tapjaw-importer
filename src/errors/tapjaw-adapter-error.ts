import TapjawAdapter from '../contracts/tapjaw-adapter';
import TapjawMessage from '../messages/tapjaw-message';
import isError from './typeguards/is-error';

export default class TapjawAdapterError extends Error {
    public parentStack?: string;

    constructor(message: unknown, adapter: TapjawAdapter<unknown, TapjawMessage>) {
        if (typeof message === 'string') {
            super(message);
            this.name = `TapjawAdapterError:${adapter.constructor.name}`;
        } else if (isError(message)) {
            super(String(message));
            this.name = `TapjawAdapterError:${adapter.constructor.name}`;
            this.parentStack = message.stack;
        } else {
            throw new Error('Unsupported message data type passed to TapjawAdapterError');
        }
    }
}
