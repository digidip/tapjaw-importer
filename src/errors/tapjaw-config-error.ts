export default class TapjawConfigError extends Error {
    constructor(message: string) {
        super(message);
        this.message =
            `\n ConfigError occured\n======================================================` + `\n${this.message}`;
        this.name = 'TapjawConfigError';
    }
}
