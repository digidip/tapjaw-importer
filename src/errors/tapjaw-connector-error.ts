import isError from '../typeguards/is-error';
import TapjawConnector from '../contracts/tapjaw-connector';

export default class TapjawConnectorError extends Error {
    public parentStack?: string;

    constructor(message: unknown, adapter: TapjawConnector) {
        if (typeof message === 'string') {
            super(message);
            this.name = `TapjawConnectorError:${adapter.constructor.name}`;
        } else if (isError(message)) {
            super(String(message));

            this.name = `TapjawConnectorError:${adapter.constructor.name}`;
            this.parentStack = message.stack;
        } else {
            throw new Error('Unsupported message data type passed to TapjawConnectorError');
        }
    }
}
