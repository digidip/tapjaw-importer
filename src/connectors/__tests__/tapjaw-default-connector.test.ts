import 'reflect-metadata';
import { createBearerSecurity, TapjawMetadata } from '../..';
import { TapjawConnectorError } from '../../errors/tapjaw-connector-error';
import TapjawDefaultConnector from '../tapjaw-default-connector';
import TapjawHttpConnector, {
    TapjawHttpQueryParameters,
    ArrayParameter,
    DuplicateParameter,
    TapjawHttpConnectorCharSet,
    TapjawHttpConnectorProtocol,
} from '../tapjaw-http-connector';

describe('Make sure TapjawDefaultConnector works as expected.', () => {
    it('should container all the configured metadata values', () => {
        @TapjawMetadata.Connector.Decode(TapjawHttpConnectorCharSet.UTF8)
        @TapjawMetadata.Connector.Encode(TapjawHttpConnectorCharSet.UTF8)
        @TapjawMetadata.Connector.Host('my.testing.com')
        @TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol.HTTPS)
        @TapjawMetadata.Connector.Port(123)
        @TapjawMetadata.Connector.EnableGzip()
        @TapjawMetadata.Connector.Security(createBearerSecurity('xxx'))
        class TestConnector extends TapjawDefaultConnector {}
        const connector = new TestConnector();

        expect(connector).toMatchSnapshot();
    });

    it('should override https port check, since .Port(80) is set, also check for latin1', () => {
        @TapjawMetadata.Connector.Decode(TapjawHttpConnectorCharSet.LATIN1)
        @TapjawMetadata.Connector.Encode(TapjawHttpConnectorCharSet.LATIN1)
        @TapjawMetadata.Connector.Host('my.moo.com')
        @TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol.HTTPS)
        @TapjawMetadata.Connector.Port(80)
        class TestConnector extends TapjawDefaultConnector {}
        const connector = new TestConnector();

        expect(connector).toMatchSnapshot();
    });

    it('should acquire port 443 since HTTPS enabled but no port is specified', () => {
        @TapjawMetadata.Connector.Decode(TapjawHttpConnectorCharSet.LATIN1)
        @TapjawMetadata.Connector.Encode(TapjawHttpConnectorCharSet.UTF8)
        @TapjawMetadata.Connector.Host('my.moo.com')
        @TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol.HTTPS)
        class TestConnector extends TapjawDefaultConnector {}
        const connector = new TestConnector();

        expect(connector).toMatchSnapshot();
    });

    it('should simple HTTP connector', () => {
        @TapjawMetadata.Connector.Decode(TapjawHttpConnectorCharSet.LATIN1)
        @TapjawMetadata.Connector.Encode(TapjawHttpConnectorCharSet.UTF8)
        @TapjawMetadata.Connector.Host('my.moo.com')
        @TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol.HTTP)
        class TestConnector extends TapjawDefaultConnector {}
        const connector = new TestConnector();

        expect(connector).toMatchSnapshot();
    });

    it('should get error because Host is missing', () => {
        @TapjawMetadata.Connector.Decode(TapjawHttpConnectorCharSet.LATIN1)
        @TapjawMetadata.Connector.Encode(TapjawHttpConnectorCharSet.UTF8)
        @TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol.HTTP)
        class TestConnector extends TapjawDefaultConnector {}

        expect(() => new TestConnector()).toThrowError(
            new TapjawConnectorError(`@TapjawMetadata.Connector.Host(string) is required!`, {
                constructor: { name: 'TapjawConnectorError' },
            } as unknown as TestConnector)
        );
    });

    it('should get error because Protocol is missing', () => {
        @TapjawMetadata.Connector.Decode(TapjawHttpConnectorCharSet.LATIN1)
        @TapjawMetadata.Connector.Encode(TapjawHttpConnectorCharSet.UTF8)
        @TapjawMetadata.Connector.Host('my.moo.com')
        class TestConnector extends TapjawDefaultConnector {}

        expect(() => new TestConnector()).toThrowError(
            new TapjawConnectorError(`@TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol) is required.`, {
                constructor: { name: 'TapjawConnectorError' },
            } as unknown as TestConnector)
        );
    });

    it('should work with minimal metadata configuration: host (HTTP) + protocol', () => {
        @TapjawMetadata.Connector.Host('my.moo.com')
        @TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol.HTTP)
        class TestConnector extends TapjawDefaultConnector {}
        const connector = new TestConnector();

        expect(connector).toMatchSnapshot();
    });

    it('should work with minimal metadata configuration: host (HTTPS) + protocol', () => {
        @TapjawMetadata.Connector.Host('my.moo.com')
        @TapjawMetadata.Connector.Protocol(TapjawHttpConnectorProtocol.HTTPS)
        class TestConnector extends TapjawDefaultConnector {}
        const connector = new TestConnector();

        expect(connector).toMatchSnapshot();
    });
});
