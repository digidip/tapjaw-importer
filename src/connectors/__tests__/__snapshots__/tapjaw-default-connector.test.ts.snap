// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure TapjawDefaultConnector works as expected. should acquire port 443 since HTTPS enabled but no port is specified 1`] = `
TestConnector {
  "enableGzip": false,
  "host": "my.moo.com",
  "lastResponse": null,
  "port": 443,
  "protocol": "https",
  "security": undefined,
  "successfulStatusCodes": [
    200,
  ],
  "useDecoding": "iso-8859-1",
  "useEncoding": "utf-8",
}
`;

exports[`Make sure TapjawDefaultConnector works as expected. should container all the configured metadata values 1`] = `
TestConnector {
  "enableGzip": true,
  "host": "my.testing.com",
  "lastResponse": null,
  "port": 123,
  "protocol": "https",
  "security": ApplyAuthorizationHttpHeaderWrapper {
    "authenticator": BearerAuthAuthenticator {
      "authenticated": false,
      "bearerToken": "xxx",
      "header": {
        "Authorization": "Bearer xxx",
      },
    },
  },
  "successfulStatusCodes": [
    200,
  ],
  "useDecoding": "utf-8",
  "useEncoding": "utf-8",
}
`;

exports[`Make sure TapjawDefaultConnector works as expected. should override https port check, since .Port(80) is set, also check for latin1 1`] = `
TestConnector {
  "enableGzip": false,
  "host": "my.moo.com",
  "lastResponse": null,
  "port": 80,
  "protocol": "https",
  "security": undefined,
  "successfulStatusCodes": [
    200,
  ],
  "useDecoding": "iso-8859-1",
  "useEncoding": "iso-8859-1",
}
`;

exports[`Make sure TapjawDefaultConnector works as expected. should simple HTTP connector 1`] = `
TestConnector {
  "enableGzip": false,
  "host": "my.moo.com",
  "lastResponse": null,
  "port": 80,
  "protocol": "http",
  "security": undefined,
  "successfulStatusCodes": [
    200,
  ],
  "useDecoding": "iso-8859-1",
  "useEncoding": "utf-8",
}
`;

exports[`Make sure TapjawDefaultConnector works as expected. should work with minimal metadata configuration: host (HTTP) + protocol 1`] = `
TestConnector {
  "enableGzip": false,
  "host": "my.moo.com",
  "lastResponse": null,
  "port": 80,
  "protocol": "http",
  "security": undefined,
  "successfulStatusCodes": [
    200,
  ],
  "useDecoding": undefined,
  "useEncoding": undefined,
}
`;

exports[`Make sure TapjawDefaultConnector works as expected. should work with minimal metadata configuration: host (HTTPS) + protocol 1`] = `
TestConnector {
  "enableGzip": false,
  "host": "my.moo.com",
  "lastResponse": null,
  "port": 443,
  "protocol": "https",
  "security": undefined,
  "successfulStatusCodes": [
    200,
  ],
  "useDecoding": undefined,
  "useEncoding": undefined,
}
`;
